### ISSUE TITLE
Authentication: `access_token` query parameter deprecated

### ISSUE THREAD

**Issue by rjbou** [mirage/ocaml-github#231]
Since 11/2019, `access_token` query parameter is [deprecated](https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters).
As it is used in opam-publish, it results on mail from github
```
On February 7th, 2020 at 17:11 (UTC) your personal access token (opam-publish
access token (xxx)) using ocaml-github ocaml-cohttp/2.4.0 was used as part of a
query parameter to access an endpoint through the GitHub API:

https://api.github.com/user

Please use the Authorization HTTP header instead, as using the `access_token`
query parameter is deprecated.

Depending on your API usage, we'll be sending you this email reminder once
every 3 days for each token and User-Agent used in API calls made on your
behalf.  Just one URL that was accessed with a token and User-Agent combination
will be listed in the email reminder, not all.

Visit
https://developer.github.com/changes/2019-11-05-deprecated-passwords-and-authorizations-api/#authenticating-using-query-parameters
for more information.
```

Related issue ocaml/opam-publish#97

**Comment by avsm** [comment]
fixed in #230, just releasing a new version now

**Comment by rjbou** [comment]
Thanks!
